src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -95,16 +95,20 @@ void CurlDeleter::operator()(CURL *curl)
  *       was needed for monitoring to report what IP protocol was being used.
  *       It has been kept in case we will need this callback in the future.
  */
-int TPCHandler::sockopt_setcloexec_callback(void *clientp, curl_socket_t curlfd, curlsocktype purpose) {
-    int oldFlags = fcntl(curlfd,F_GETFD,0);
+int TPCHandler::sockopt_callback(void *clientp, curl_socket_t curlfd, curlsocktype purpose) {
+  if(purpose == curlsocktype::CURLSOCKTYPE_IPCXN) {
+    return CURL_SOCKOPT_ALREADY_CONNECTED;
+  }
+  return CURL_SOCKOPT_OK;
+    /*int oldFlags = fcntl(curlfd,F_GETFD,0);
     if(oldFlags < 0) {
         return CURL_SOCKOPT_ERROR;
     }
     oldFlags |= FD_CLOEXEC;
     if(!fcntl(curlfd,F_SETFD,oldFlags)) {
         return CURL_SOCKOPT_OK;
     }
-    return CURL_SOCKOPT_ERROR;
+    return CURL_SOCKOPT_ERROR;*/
```

Comment:
Why leave commented out code in? I think it's ok to remove and refer to the git log if necessary.

Reply:
I would say please remove it, it's confusing the way it is.